packages/pds-ember/tests/integration/components/checkbox-field/index-test.js

```diff
@@ -0,0 +1,84 @@
+import { module, test } from 'qunit'
+import { setupRenderingTest } from 'ember-qunit'
+import { render, settled } from '@ember/test-helpers'
+import { hbs } from 'ember-cli-htmlbars'
+
+const ROOT = '[data-test-checkbox-field]'
+const CONTROL = '[data-test-checkbox-field-control]'
+
+module('Integration | Components.CheckboxField', function(hooks) {
+  setupRenderingTest(hooks)
+
+  test('it renders', async function(assert) {
+    await render(hbs`
+      <Pds::CheckboxField
+        data-foo="bar"
+      />
+    `)
+
+    assert
+      .dom(ROOT)
+      .exists()
+      .hasNoText()
+      .hasClass('pds-checkboxField')
+
+    assert
+      .dom(CONTROL)
+      .exists()
+      .hasAttribute('data-foo', 'bar', 'applies ...attributes')
+
+    // Template block usage:
+    await render(hbs`
+      <Pds::CheckboxField>
+        template block text
+      </Pds::CheckboxField>
+    `)
+
+    assert
+      .dom(ROOT)
+      .hasText('template block text')
+  })
+
+  test('it yields an auto-generated id', async function(assert) {
+    await render(hbs`
+      <Pds::CheckboxField as |f|>
+        {{f.id}}
+      </Pds::CheckboxField>
+    `)
+
+    assert
+      .dom(ROOT)
+      .hasText(/chk-ember\d+/, 'yields generated id')
+  })
+
+  test('it favors [id] over auto-generated id', async function(assert) {
```

Comment:
The `[id]` attribute should be avoided, because it doesn't reflect to the generated `label[for]` attribute.  Need to update the component to support `@id` in order to do so.